File: src/main/java/br/com/senac/rentacar/entities/Locacao.java
package br.com.senac.rentacar.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Locacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public LocalDate dataInicio;

    public LocalDate dataFim;

    public Double valorTotal;

    public String cliente;

    public String veiculo;
}

File: frontn/app/types/locacao.ts
import { Cliente } from "./cliente";
import { Veiculo } from "./veiculo";

export class Locacao{

constructor (

    public id: number | null,
    public dataInicio:string,
    public dataFim:string,
    public valorTotal:number,
    public cliente:Cliente | null,
    public veiculo:Veiculo | null,

    ){}

}